scripts/globals/extravaganza.lua

```diff
@@ -0,0 +1,73 @@
+-----------------------------------
+-- Alter Ego Extravaganza
+-----------------------------------
+require('scripts/globals/settings')
+require('scripts/globals/items')
+require('scripts/globals/npc_util')
+-----------------------------------
+
+xi = xi or {}
+xi.extravaganza = {}
+
+local enabled = xi.settings.main.ENABLE_TRUST_ALTER_EGO_EXTRAVAGANZA
+
+-----------------------------------------------
+-- Determine Which Ciphers to Grant by NPC Name
+-----------------------------------------------
+
+xi.extravaganza.getShadowEraCiphers = function(npc)
+    local ciphers = {}
+    switch (npc:getName()): caseof
```

Comment:
No need to use a switch here.  This could be defined as a keyed table:
```
local  cipherNpc =
{
    ['Shixo'] = { item1, item2 },
    ...
}

return cipherNpc[npc:getName()]
```